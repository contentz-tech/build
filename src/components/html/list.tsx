import { jsx } from "@emotion/core";

function UL(props: any) {
  return (
    <ul
      {...props}
      css={{
        fontSize: "1em",
        fontWeight: "normal",
        marginLeft: "-2rem",
        paddingLeft: "2rem",
        listStyleType: "square",
        "@media (max-width: 767px)": {
          marginLeft: 0,
          paddingLeft: "2rem"
        },
        ul: {
          paddingLeft: "3.25rem"
        }
      }}
    />
  );
}

function OL(props: any) {
  return (
    <ol
      {...props}
      css={{
        fontSize: "1em",
        fontWeight: "normal",
        marginLeft: "-2rem",
        paddingLeft: "2rem",
        "@media (max-width: 767px)": {
          marginLeft: 0,
          paddingLeft: "2rem"
        },
        ol: {
          paddingLeft: "3.25rem"
        }
      }}
    />
  );
}

function LI(props: any) {
  return <li {...props} css={{ lineHeight: 1.3 }} />;
}

export { UL, OL, LI };
