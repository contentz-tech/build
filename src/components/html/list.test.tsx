/** * @jest-environment jsdom */
import '@testing-library/jest-dom/extend-expect'
import { jsx } from "@emotion/core";
import { render } from "@testing-library/react";
import { UL, OL, LI } from "./list";

describe("List", () => {
  test("Ordered", () => {
    const { asFragment, getByText } = render(
      <OL>
        <LI>Item 1</LI>
        <LI>Item 2</LI>
        <LI>Item 3</LI>
      </OL>
    );

    expect(asFragment()).toMatchSnapshot();

    expect(getByText(/Item 1/)).toHaveTextContent("Item 1");
    expect(getByText(/Item 2/)).toHaveTextContent("Item 2");
    expect(getByText(/Item 3/)).toHaveTextContent("Item 3");
  });

  test("Unordered", () => {
    const { asFragment, getByText } = render(
      <UL>
        <LI>Item 1</LI>
        <LI>Item 2</LI>
        <LI>Item 3</LI>
      </UL>
    );

    expect(asFragment()).toMatchSnapshot();

    expect(getByText(/Item 1/)).toHaveTextContent("Item 1");
    expect(getByText(/Item 2/)).toHaveTextContent("Item 2");
    expect(getByText(/Item 3/)).toHaveTextContent("Item 3");
  });
});
