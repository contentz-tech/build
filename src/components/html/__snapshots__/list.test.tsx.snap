// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Ordered 1`] = `
<DocumentFragment>
  .emotion-3 {
  font-size: 1em;
  font-weight: normal;
  margin-left: -2rem;
  padding-left: 2rem;
}

@media (max-width:767px) {
  .emotion-3 {
    margin-left: 0;
    padding-left: 2rem;
  }
}

.emotion-3 ol {
  padding-left: 3.25rem;
}

.emotion-0 {
  line-height: 1.3;
}

<ol
    class="emotion-3"
  >
    <li
      class="emotion-0"
    >
      Item 1
    </li>
    <li
      class="emotion-0"
    >
      Item 2
    </li>
    <li
      class="emotion-0"
    >
      Item 3
    </li>
  </ol>
</DocumentFragment>
`;

exports[`List Unordered 1`] = `
<DocumentFragment>
  .emotion-0 {
  line-height: 1.3;
}

.emotion-3 {
  font-size: 1em;
  font-weight: normal;
  margin-left: -2rem;
  padding-left: 2rem;
  list-style-type: square;
}

@media (max-width:767px) {
  .emotion-3 {
    margin-left: 0;
    padding-left: 2rem;
  }
}

.emotion-3 ul {
  padding-left: 3.25rem;
}

<ul
    class="emotion-3"
  >
    <li
      class="emotion-0"
    >
      Item 1
    </li>
    <li
      class="emotion-0"
    >
      Item 2
    </li>
    <li
      class="emotion-0"
    >
      Item 3
    </li>
  </ul>
</DocumentFragment>
`;
